docs/architecture/adr-110-e2e-abci++.md

```diff
@@ -0,0 +1,267 @@
+# ADR 82: E2E tests for CometBFT's behaviour in respect to ABCI++.
```

Comment:
Please check what we right number is (CometBFT started at 100) by looking at `docs/architecture/README.md`

Reply:
Also, rename the file accordingly